internal/provider/provider.go

```diff
@@ -201,23 +212,40 @@ func providerConfigure(st state.StateI) (func(ctx context.Context, d *schema.Res
 		}
 		apiClient.twin_id = twin
 		var cl rmb.Client
-		if apiClient.use_rmb_proxy {
-			verify_reply := d.Get("verify_reply").(bool)
-			cl, err = client.NewProxyBus(rmb_proxy_url, apiClient.twin_id, apiClient.substrateConn, identity, verify_reply)
+
+		sessionID := generateSessionID(twin)
+
+		sub, err := apiClient.manager.Substrate()
+		if err != nil {
+			return nil, diag.FromErr(errors.Wrap(err, "failed to get substrate client"))
+		}
+		relayURL := d.Get("relay_url").(string)
+		if relayURL != "" {
+			cl, err = direct.NewClient(key_type, apiClient.mnemonics, relayURL, sessionID, sub)
 		} else {
-			cl, err = rmb.NewRMBClient(apiClient.rmb_redis_url)
+			relayURL, ok := RELAYS_URLS[network]
+			if !ok {
+				return nil, diag.Errorf("error getting relay url for network %s", network)
+			}
+
+			cl, err = direct.NewClient(key_type, apiClient.mnemonics, relayURL, sessionID, sub)
 		}
+
 		if err != nil {
 			return nil, diag.FromErr(errors.Wrap(err, "couldn't create rmb client"))
 		}
 		apiClient.rmb = cl
 
 		grid_client := proxy.NewClient(rmb_proxy_url)
 		apiClient.grid_client = proxy.NewRetryingClient(grid_client)
-		if err := preValidate(&apiClient, sub); err != nil {
+		if err := preValidate(&apiClient, subx); err != nil {
 			return nil, diag.FromErr(err)
 		}
 		apiClient.state = st
 		return &apiClient, nil
 	}, substrateConn
 }
+
+func generateSessionID(twinID uint32) string {
```

Comment:
where is twinID used?

Reply:
after several modifications, it's not used and should be removed for sure.